Makefile

```diff
@@ -24,6 +24,32 @@ LDFLAGS ?= "-X $(BUILD_DATE_VAR)=$(BUILD_DATE) -X $(BUILD_VERSION_VAR)=$(VERSION
 E2E_FLAGS ?= -installType=KindCluster
 E2E_FLAGS_DEFAULT := -test.v -ginkgo.v -ginkgo.progress -ctrRegistry $(CTR_REGISTRY) -osmImageTag $(CTR_TAG)
 
+# Reported Go version
+GO_VERSION_MAJOR = $(shell go version | grep -Po 'go\K(\d+)')
+GO_VERSION_MINOR = $(shell go version | grep -Po 'go\d+\.\K(\d+)')
+GO_VERSION_PATCH = $(shell go version | grep -Po 'go\d+\.\d+\.\K(\d+)')
+
+# Required Go version
+MIN_REQUIRED_GO_VERSION_MAJOR = 1
+MIN_REQUIRED_GO_VERSION_MINOR = 15
+MIN_REQUIRED_GO_VERSION_PATCH = 7
```

Comment:
Can we document why the reason for requiring this specific version at a minimum?

Reply:
Why does dev/test require v1.15.7? Isn't the potential issue in e2e and CI already uses latest v1.15 to build and publish?

Reply:
@nojnhuh @sozercan - added documentation here - let me know if you think this adds clarity.